packages/mongodb-chatbot-server/src/services/ChatLlm.ts

```diff
@@ -45,20 +52,44 @@ export interface Tool {
   /**
     Call the function based on the arguments in the {@link Tool.definition}.
    */
-  call(args: unknown): Promise<CallToolResponse>;
+  call(args: ToolCallParams): Promise<CallToolResponse>;
+}
+
+export interface ToolCallParams<T = unknown> {
```

Comment:
Avoid making parameter interfaces generic, as the generic element won't actually be usable by the consuming function. (To see what I mean, note that you would have to make LlmAnswerQuestionParams generic and propagate a T from there into `call(args: ToolCallParams<T>)` in order for this to work... and following that chain, you'll eventually make everything generic, only to realize it was meant to be concrete the whole time...)
```suggestion
export interface ToolCallParams {
  // functionArgs: unknown;
```